vizro-ai/examples/dashboard_ui/components.py

```diff
@@ -0,0 +1,214 @@
+                        "lineHeight": "45px",
+                        "borderWidth": "1px",
+                        "borderStyle": "dashed",
+                        "borderRadius": "5px",
+                        "textAlign": "center",
+                        "color": "#d1d1d5",
+                        "borderColor": "rgba(43, 46, 57, 1)",
+                    },
+                ),
+            ]
+        )
+
+
+class CodeClipboard(vm.VizroBaseModel):
+    """Code snippet with a copy to clipboard button."""
+
+    type: Literal["code_clipboard"] = "code_clipboard"
+    code: str = ""
+    language: str = "python"
+
+    def build(self):
+        """Returns the code clipboard component inside a output text area."""
+        code = black.format_str(self.code, mode=black.Mode(line_length=120))
+        code = code.strip("'\"")
+
+        markdown_code = "\n".join(["```python", code, "```"])
+
+        return html.Div(
+            [
+                dcc.Clipboard(target_id=f"{self.id}-code-markdown", className="code-clipboard"),
+                dcc.Markdown(markdown_code, id=f"{self.id}-code-markdown"),
+            ],
+            className="code-clipboard-container",
+        )
+
+
+class MyDropdown(vm.Dropdown):
+    """Custom dropdown component."""
+
+    type: Literal["my_dropdown"] = "my_dropdown"
+
+    def build(self):
+        """Returns custom dropdown component that cannot be cleared."""
+        dropdown_build_obj = super().build()
+        dropdown_build_obj.id = f"{self.id}_outer_div"
+        dropdown_build_obj.children[1].clearable = False
+
+        return dropdown_build_obj
+
+
+class OffCanvas(vm.VizroBaseModel):
+    """OffCanvas component for settings."""
+
+    type: Literal["offcanvas"] = "offcanvas"
+    options: List[str]
+    value: str
+
+    def build(self):
+        """Returns the off canvas component for settings."""
+        input_groups = html.Div(
+            [
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("API Key"),
+                        dbc.Input(placeholder="API key", type="password", id=f"{self.id}-api-key"),
+                    ],
+                    className="mb-3",
+                ),
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("API base"),
+                        dbc.Input(placeholder="API base", type="password", id=f"{self.id}-api-base"),
+                    ],
+                    className="mb-3",
+                ),
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("Choose your vendor"),
+                        dbc.Select(options=self.options, value=self.value, id=f"{self.id}-dropdown"),
+                    ],
+                    className="mb-3",
+                ),
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText(dbc.Checkbox(id=f"{self.id}-show-secrets-id")),
```

Comment:
Can you remind me why toggle not working in offcanvas? Definitely not an issue using Checkbox, just wondering why we cannot use toggle-like switches
```
switches = html.Div(
    [
        dbc.Label("Toggle a bunch"),
        dbc.Checklist(
            options=[
                {"label": "Option 1", "value": 1},
                {"label": "Option 2", "value": 2},
                {"label": "Disabled Option", "value": 3, "disabled": True},
            ],
            value=[1],
            id="switches-input",
            switch=True,
        ),
    ]
)